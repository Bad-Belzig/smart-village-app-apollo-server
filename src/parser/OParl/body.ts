import { isObjectLike } from 'lodash';
import { parseDate } from '../dateParser';
import { mapToIds } from '../helpers';

// only the necessary validation for parsing is done here.
// the created database object will be validated after creation before saving
// by using the .validate method that checks against the provided schema
export const parseBody = (json) => {
  if (isObjectLike(json)) {
    const externalId = json.id; // change the id given to us to be the externalId instead
    delete json.id; // remove the id key from the json to ensure we do not preset it for mongo db
    return {
      ...json,
      externalId,
      created: parseDate(json.created),

      // filter out objects that have no id
      legislativeTerm: mapToIds(json.legislativeTerm),
      modified: parseDate(json.modified),
      licenseValidSince: parseDate(json.licenseValidSince),
      oparlSince: parseDate(json.oparlSince),
    };
  }
};
