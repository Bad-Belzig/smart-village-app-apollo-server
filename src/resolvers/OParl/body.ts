import {
  Body,
  IBody,
  LegislativeTerm,
  Location,
  Meeting,
  Organization,
  Paper,
  Person,
} from '../../models';

export const bodyResolvers = {
  Query: {
    oParlBodies: () => Body.find(),
  },
  OParlBody: {
    organization: (args: IBody) =>
      args.organization.map((value) =>
        Organization.findOne({ externalId: value }),
      ),
    person: (args: IBody) =>
      args.person.map((value) => Person.findOne({ externalId: value })),
    meeting: (args: IBody) =>
      args.meeting.map((value) => Meeting.findOne({ externalId: value })),
    paper: (args: IBody) =>
      args.paper.map((value) => Paper.findOne({ externalId: value })),
    legislativeTerm: (args: IBody) =>
      args.legislativeTerm.map((value) =>
        LegislativeTerm.findOne({ externalId: value }),
      ),
    location: (args: IBody) => Location.findOne({ externalId: args.location }),
  },
};
