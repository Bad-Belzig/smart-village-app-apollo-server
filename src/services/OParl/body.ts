import { Body } from '../../models';
import { parseBody } from '../../parser';
import { getJson, updateOrCreateEntry } from './importHelpers';
import { ImportQueue, ImportQueueEntry, ImportType } from './ImportTypes';
import { fetchPaginatedOParlList } from './paginatedList';

export const importBody = async (
  value: unknown,
  queue?: ImportQueue,
  createdSince?: Date,
  createdUntil?: Date,
) => {
  const json = await getJson(value);

  if (!json) return;

  const addToQueue: [ImportQueueEntry | ImportQueueEntry[], ImportType][] = [];

  if (json.legislativeTerm) {
    addToQueue.push([json.legislativeTerm, ImportType.LegislativeTerm]);
  }

  if (json.system) {
    addToQueue.push([json.system, ImportType.System]);
  }

  if (json.meeting) {
    json.meeting = await fetchPaginatedOParlList(
      json.meeting,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.meeting, ImportType.Meeting]);
  }

  if (json.organization) {
    json.organization = await fetchPaginatedOParlList(
      json.organization,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.organization, ImportType.Organization]);
  }

  if (json.paper) {
    json.paper = await fetchPaginatedOParlList(
      json.paper,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.paper, ImportType.Paper]);
  }

  if (json.person) {
    json.person = await fetchPaginatedOParlList(
      json.person,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.person, ImportType.Person]);
  }

  if (json.agendaItem) {
    json.agendaItem = await fetchPaginatedOParlList(
      json.agendaItem,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.agendaItem, ImportType.AgendaItem]);
  }

  if (json.consultation) {
    json.consultation = await fetchPaginatedOParlList(
      json.consultation,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.consultation, ImportType.Consultation]);
  }

  if (json.file) {
    json.file = await fetchPaginatedOParlList(
      json.file,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.file, ImportType.File]);
  }

  if (json.legislativeTermList) {
    json.legislativeTermList = await fetchPaginatedOParlList(
      json.legislativeTermList,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.legislativeTermList, ImportType.LegislativeTerm]);
  }

  if (json.locationList) {
    json.locationList = await fetchPaginatedOParlList(
      json.locationList,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.locationList, ImportType.Location]);
  }

  if (json.membership) {
    json.membership = await fetchPaginatedOParlList(
      json.membership,
      createdSince,
      createdUntil,
    );
    addToQueue.push([json.membership, ImportType.Membership]);
  }

  try {
    return await updateOrCreateEntry(json, parseBody, Body, addToQueue, queue);
  } catch (e) {
    console.log(`Error while importing body from ${value}:`, e);
  }
};
