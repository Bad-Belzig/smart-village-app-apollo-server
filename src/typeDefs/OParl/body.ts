import { gql } from 'apollo-server-core';

// TODO: this needs to be refined once the other types are integrated and we can nest the data instead of having link references"
export const bodyGQL = gql`
  extend type Query {
    oParlBodies: [OParlBody!]
  }

  extend type Mutation {
    updateBody(externalId: String!, name: String!): UpdateBodyResponse
  }

  type OParlBody {
    id: ID!
    "id from the oparl object = url where the json is located"
    externalId: String!
    "oParl type = url where the type is specified"
    type: String!
    created: Date
    modified: Date
    keyword: [String!]
    web: String
    deleted: Boolean
    name: String!
    website: String
    license: String
    licenseValidSince: Date
    oparlSince: Date
    ags: String
    rgs: String
    equivalent: [String!]
    contactEmail: String
    contactName: String
    organization: [OParlOrganization!]
    person: [OParlPerson]
    meeting: [OParlMeeting!]
    paper: [OParlPaper!]
    legislativeTerm: [OParlLegislativeTerm!]
    classification: String
    location: [OParlLocation!]
  }

  "TODO: either remove the ability to mutate oparl object via graphql or refine it"
  type UpdateBodyResponse {
    body: OParlBody
    success: Boolean
  }
`;
