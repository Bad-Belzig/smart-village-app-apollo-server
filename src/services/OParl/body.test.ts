import {
  basicImportTest,
  invalidUrlTest,
  setupTestDataBase,
} from '../../helpers';
import { importBody } from './body';

const helperFuncs = setupTestDataBase('test_body_import');

// TODO: change test url to something within our control
const testUrl =
  'https://oparl.politik-bei-uns.de/body/5a73013af24bb7ed5127a72a';
const testUrl2 =
  'https://sdnetrim.kdvz-frechen.de/rim4883/webservice/oparl/v1.1/body/1';

describe('importing a body', () => {
  beforeAll(async () => {
    await helperFuncs.beforeAll();
    jest.setTimeout(30000);
  });

  it(
    'imports properly for 1.0',
    basicImportTest(
      importBody,
      testUrl,
      new Date('2017-05-01'),
      new Date('2017-05-31'),
    ),
  );

  it(
    'imports properly for 1.1',
    basicImportTest(
      importBody,
      testUrl2,
      new Date('2016-06-02'),
      new Date('2016-07-31'),
    ),
  );

  it('import returns undefined on faulty url', invalidUrlTest(importBody));

  afterAll(async () => {
    await helperFuncs.afterAll();
    jest.setTimeout(5000);
  });
});
