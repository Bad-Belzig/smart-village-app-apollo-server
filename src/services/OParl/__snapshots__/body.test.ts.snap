// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importing a body imports properly 1`] = `
Object {
  "__v": 0,
  "ags": "09162000",
  "contactEmail": "info@muenchen-transparent.de",
  "contactName": "München Transparent",
  "created": 2016-10-16T18:38:18.000Z,
  "equivalent": Array [],
  "externalId": "https://www.muenchen-transparent.de/oparl/v1.0/body/0",
  "keyword": Array [],
  "legislativeTerm": Array [
    "https://www.muenchen-transparent.de/oparl/v1.0/legislativeterm/0",
    "https://www.muenchen-transparent.de/oparl/v1.0/legislativeterm/1",
    "https://www.muenchen-transparent.de/oparl/v1.0/legislativeterm/2",
    "https://www.muenchen-transparent.de/oparl/v1.0/legislativeterm/3",
    "https://www.muenchen-transparent.de/oparl/v1.0/legislativeterm/4",
  ],
  "meeting": Array [
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4090690",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4090779",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4090875",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4090962",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136050",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136133",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136178",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136294",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136412",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136450",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136488",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4136647",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4143940",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4200408",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4205238",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4207647",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209129",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209167",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209354",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209387",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209423",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209509",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209622",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4209651",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4218929",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4269943",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4270023",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4270147",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4270519",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4288587",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4288627",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4288656",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4313499",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4347061",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4368030",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4368147",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4368220",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4368362",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4404147",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4404184",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4404220",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4404306",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4404388",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4434753",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4434835",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4434918",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4435722",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4435761",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4482429",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4482465",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4482831",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4502026",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4503495",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4503544",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4503589",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4503662",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4503772",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4532747",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4532786",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4532824",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4532896",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4532940",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4572503",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4572596",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4572868",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4572940",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4582750",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4582829",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4622441",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4622480",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4653822",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4653923",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4653996",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4654567",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4654686",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4654927",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4679723",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4682474",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690299",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690339",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690494",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690531",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690569",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690605",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690649",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690720",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690752",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4690799",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4691483",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4701256",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4701325",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4701367",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4706927",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4707360",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4710530",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4711972",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4740218",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4740253",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4740345",
    "https://www.muenchen-transparent.de/oparl/v1.0/meeting/4740791",
  ],
  "modified": 2016-10-16T18:39:16.000Z,
  "name": "Stadrat der Landeshauptstadt München",
  "organization": Array [
    "https://www.muenchen-transparent.de/oparl/v1.0/organization/referat/14",
    "https://www.muenchen-transparent.de/oparl/v1.0/organization/gremium/219",
    "https://www.muenchen-transparent.de/oparl/v1.0/organization/gremium/4769053",
    "https://www.muenchen-transparent.de/oparl/v1.0/organization/fraktion/4203027",
  ],
  "paper": Array [
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/1505496",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/2293143",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/2739707",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/2806759",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/2941300",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/2972553",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/2997009",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3307941",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3452825",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3627952",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3673363",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3713729",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3733023",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3771424",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3778170",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3779919",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3780470",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3821649",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3911050",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3945285",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/3965186",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4003922",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4004012",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4012715",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4013070",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4026354",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4039162",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4044165",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4044744",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4049176",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4055817",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4056214",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4058724",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4061818",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4063942",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4064717",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4069575",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4086361",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4110913",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4111627",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4122834",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4138686",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4139390",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4151894",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4154645",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4161531",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4171576",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4176927",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4179351",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4183935",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4190812",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4203886",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4213777",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4215142",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4225461",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4245795",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4246831",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4261497",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4269849",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4273560",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4279463",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4281307",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4291099",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4291325",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4295653",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4298388",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4301969",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4305100",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4305788",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4305867",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4307737",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4308427",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4309183",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4309880",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4316023",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4316218",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4316265",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4316297",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4317570",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4324154",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4327967",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4331662",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4337090",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4337123",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4337307",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4337431",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4337555",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4339825",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4340715",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4340925",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4341328",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4341785",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4342218",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4342436",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4342513",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4342598",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4343592",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4344273",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4344281",
    "https://www.muenchen-transparent.de/oparl/v1.0/paper/4344348",
  ],
  "person": Array [
    "https://www.muenchen-transparent.de/oparl/v1.0/person/476",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/485",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/544",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/545",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/556",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/575",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/796",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/809",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/944",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/953",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/660913",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/773029",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/1133486",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/1235663",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/1419253",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/1419651",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/1420214",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/1421418",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/1440165",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/2076312",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/2778515",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/2952619",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/3584862",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/3638421",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/3721038",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4023435",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4024415",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4026084",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4058022",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4062696",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4066092",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4066096",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4097996",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4098046",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4098057",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4114963",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4115708",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4115713",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4115845",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4159830",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4193170",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4193172",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4193179",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4235947",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4237097",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4237104",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4237140",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4238339",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4245999",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4246418",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4283719",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4295592",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4313166",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4321901",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4322579",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4350633",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4376267",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4377076",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4397156",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4412819",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4427027",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4434394",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4444245",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4454539",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4454694",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4485974",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4485985",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4486009",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4508979",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4509832",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4511489",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4549906",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4551064",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4566073",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4567124",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4577568",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4609714",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4609748",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4637718",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4669514",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4693794",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4705253",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4727594",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4744651",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4800210",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4800265",
    "https://www.muenchen-transparent.de/oparl/v1.0/person/4873150",
  ],
  "rgs": "606109162000",
  "type": "https://schema.oparl.org/1.0/Body",
  "web": "https://www.muenchen-transparent.de/",
  "website": "https://www.muenchen.de/rathaus/",
}
`;
